tests/std/tests/P2374R4_views_cartesian_product/test.cpp

```diff
@@ -0,0 +1,1030 @@
+        {3, 22}, {3, 33}, {4, 11}, {4, 22}, {4, 33}}),
+
+    // Expected result of views::cartesian_product(get<0>(some_ranges), ..., get<2>(some_ranges))
+    to_array<tuple<int, int, char>>(
+        {{0, 11, '7'}, {0, 22, '7'}, {0, 33, '7'}, {1, 11, '7'}, {1, 22, '7'}, {1, 33, '7'}, {2, 11, '7'}, {2, 22, '7'},
+            {2, 33, '7'}, {3, 11, '7'}, {3, 22, '7'}, {3, 33, '7'}, {4, 11, '7'}, {4, 22, '7'}, {4, 33, '7'}}),
+
+    // Expected result of views::cartesian_product(get<0>(some_ranges), ..., get<3>(some_ranges))
+    to_array<tuple<int, int, char, string_view>>(
+        {{0, 11, '7', "4"sv}, {0, 11, '7', "2"sv}, {0, 11, '7', "0"sv}, {0, 22, '7', "4"sv}, {0, 22, '7', "2"sv},
+            {0, 22, '7', "0"sv}, {0, 33, '7', "4"sv}, {0, 33, '7', "2"sv}, {0, 33, '7', "0"sv}, {1, 11, '7', "4"sv},
+            {1, 11, '7', "2"sv}, {1, 11, '7', "0"sv}, {1, 22, '7', "4"sv}, {1, 22, '7', "2"sv}, {1, 22, '7', "0"sv},
+            {1, 33, '7', "4"sv}, {1, 33, '7', "2"sv}, {1, 33, '7', "0"sv}, {2, 11, '7', "4"sv}, {2, 11, '7', "2"sv},
+            {2, 11, '7', "0"sv}, {2, 22, '7', "4"sv}, {2, 22, '7', "2"sv}, {2, 22, '7', "0"sv}, {2, 33, '7', "4"sv},
+            {2, 33, '7', "2"sv}, {2, 33, '7', "0"sv}, {3, 11, '7', "4"sv}, {3, 11, '7', "2"sv}, {3, 11, '7', "0"sv},
+            {3, 22, '7', "4"sv}, {3, 22, '7', "2"sv}, {3, 22, '7', "0"sv}, {3, 33, '7', "4"sv}, {3, 33, '7', "2"sv},
+            {3, 33, '7', "0"sv}, {4, 11, '7', "4"sv}, {4, 11, '7', "2"sv}, {4, 11, '7', "0"sv}, {4, 22, '7', "4"sv},
+            {4, 22, '7', "2"sv}, {4, 22, '7', "0"sv}, {4, 33, '7', "4"sv}, {4, 33, '7', "2"sv}, {4, 33, '7', "0"sv}}),
+};
+
+template <test::CanDifference Difference>
+struct test_input_range {
+    template <class ValTy>
+    using type = test::range<input_iterator_tag, ValTy, test::Sized::yes, Difference, test::Common::yes,
+        test::CanCompare::yes, test::ProxyRef::no>;
+};
+
+template <class Category, test::Common IsCommon, test::Sized IsSized>
+struct test_range {
+    template <class ValTy>
+    using type =
+        test::range<Category, ValTy, IsSized, test::CanDifference{derived_from<Category, random_access_iterator_tag>},
+            IsCommon, test::CanCompare{derived_from<Category, forward_iterator_tag> || IsCommon == test::Common::yes},
+            test::ProxyRef{!derived_from<Category, contiguous_iterator_tag>}>;
+};
+
+struct instantiator {
+    template <class R>
+    static constexpr void call() {
+        typename R::template type<const int> r0{get<0>(some_ranges)};
+        test_one(get<0>(expected_results), r0);
+
+        if constexpr (ranges::forward_range<typename R::template type<const int>>) {
+            typename R::template type<const int> r1{get<1>(some_ranges)};
+            test_one(get<1>(expected_results), r0, r1);
+
+#if !(defined(__clang__) && defined(_DEBUG) && defined(_MT) && !defined(_DLL)) // constexpr limit

```

Comment:
`defined(_MT)` is a red herring here (it's always defined; a relic of when we had single-threaded and multi-threaded CRTs).